### PR TITLE
fix: search terms highlighted in data-explorer

### LINKED ISSUE #541: [BUG]: Search terms not highlighted in snippits
**Describe the bug**
Search terms are not highlighted in the snippits

**To Reproduce**
search for a term, go to results with snippits, search terms are not highlighted

**Expected behavior**
Search terms should be highlighted in the snippits

**Screenshots**
image

**Additional context**

### PR DESCRIPTION
**Pull request by YohannParis**
# Description

* Added capabilities to highligh terms on the search results
* Removing the button to switch to matrix view
* Updating the highlight colour
* fix some issues with missing data handling

Resolves #541 

# Screenshots

video

### CODE REVIEW

File: packages/client/hmi-client/src/views/DataExplorer.vue [hunk 1]
```diff
@@ -437,18 +411,22 @@ const toggleDataItemSelected = (dataItem: { item: ResultType; type?: string }) =
 	let foundIndx = -1;
 	const item = dataItem.item;
 
+	console.log({
+		item,
+		foundIndx
+	});
+
```
Comment (shawnyama) [inline packages/client/hmi-client/src/views/DataExplorer.vue:418]: console log

File: packages/client/hmi-client/src/views/DataExplorer.vue [hunk 2]
```diff
@@ -581,20 +559,16 @@ onUnmounted(() => {
...
+	background-color: var(--surface-0);
 }
 
 .p-buttonset button[active='true'],
 .p-button.p-button-text:enabled:focus {
 	font-weight: bold;
 	border: none;
-	background-color: var(--primary-color-lighter);
+	background-color: var(--surface-highlight);
 }
```
Comment (shawnyama) [inline packages/client/hmi-client/src/views/DataExplorer.vue:573]: Is it okay that the selected button doesn't have a border, looks a bit off

Comment (YohannParis) [reply packages/client/hmi-client/src/views/DataExplorer.vue:573]: I'm working on redesigning the buttons at the theme level, we shouldn't have to design them on a view.

Comment (echl) [reply packages/client/hmi-client/src/views/DataExplorer.vue:573]: Agree it looks a bit odd

File: packages/client/hmi-client/src/components/data-explorer/asset-card.vue [hunk 3]
```diff
@@ -239,7 +256,6 @@ const formatFeatures = () => {
 	color: var(--text-color-subdued);
 	padding: 1rem;
 	margin: 1px;
-	font-size: 12px;
 	display: flex;
 	justify-content: space-between;
 }
```
[the remark this answers is not in the document]
Comment (YohannParis) [reply packages/client/hmi-client/src/components/data-explorer/asset-card.vue:261]: Nice I added new CSS properties wth @jamiewaese-uncharted to deal with those hard-coded values:
```scss
--font-caption: $fontCaption;
    --font-body-small: $fontBodySmall;
    --font-body-medium: $fontBodyMedium;
    --font-body-large: $fontBodyLarge;
    --font-size: $fontSize;
    --font-weight: $fontWeight;
    --font-weight-semibold: $fontWeightSemiBold;
```

File: packages/client/hmi-client/src/components/data-explorer/search-item.vue [hunk 4]
```diff
@@ -34,17 +35,18 @@ const isSelected = () =>
...
 		:asset="asset"
 		:resourceType="resourceType"
 		:active="isPreviewed"
+		:highlight="searchTerm"
 		@click="emit('toggle-asset-preview')"
 	>
 		<button type="button" @click.stop="emit('toggle-selected-asset')">
-			<i class="pi pi-plus" v-show="!isSelected()"></i>
-			<i class="pi pi-check checkmark-color" v-show="isSelected()"></i>
+			<i class="pi pi-plus" v-show="!isSelected()" />
```
Comment (echl) [inline packages/client/hmi-client/src/components/data-explorer/search-item.vue:46]: Just curious, is there any difference between doing this and doing ```<button icon=pi pi-plus>``` ?

Comment (YohannParis) [reply packages/client/hmi-client/src/components/data-explorer/search-item.vue:46]: It shouldn't

**Review by shawnyama — APPROVED** [review]

File: packages/client/hmi-client/src/components/data-explorer/search-item.vue (hunk 4 above)
Comment (echl) [reply packages/client/hmi-client/src/components/data-explorer/search-item.vue:46]: Makes sense!

**Review by echl — DISMISSED** [review]

**Review by mj3cheun — APPROVED** [review]

**Pull request merged**